pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -999,10 +1004,21 @@ class Timedelta(_Timedelta):
         if isinstance(value, Timedelta):
             value = value.value
         elif is_string_object(value):
-            if len(value) > 0 and value[0] == 'P':
+            # Check if it is just a number in a string
+            try:
```

Comment:
just try to float here. if it succeeds it *must* have a unit specified, if not you must raise, e.g. ``Timedelta('1000')`` is not accepted